radix-engine-interface/src/blueprints/epoch_manager/invocations.rs

```diff
@@ -75,53 +74,59 @@ impl Clone for EpochManagerCreateInput {
 
 pub const EPOCH_MANAGER_GET_CURRENT_EPOCH_IDENT: &str = "get_current_epoch";
 
-#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
+#[derive(Debug, Clone, Eq, PartialEq, Sbor)]
 pub struct EpochManagerGetCurrentEpochInput;
 
 pub const EPOCH_MANAGER_SET_EPOCH_IDENT: &str = "set_epoch";
 
-#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
+#[derive(Debug, Clone, Eq, PartialEq, Sbor)]
 pub struct EpochManagerSetEpochInput {
     pub epoch: u64,
 }
 
 pub const EPOCH_MANAGER_NEXT_ROUND_IDENT: &str = "next_round";
 
-#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
+#[derive(Debug, Clone, Eq, PartialEq, Sbor)]
 pub struct EpochManagerNextRoundInput {
     pub round: u64,
 }
 
 pub const EPOCH_MANAGER_CREATE_VALIDATOR_IDENT: &str = "create_validator";
 
-#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
+#[derive(
+    Debug, Clone, Eq, PartialEq, ScryptoSbor, ManifestCategorize, ManifestEncode, ManifestDecode,
+)]
 pub struct EpochManagerCreateValidatorInput {
     pub key: EcdsaSecp256k1PublicKey,
     pub owner_access_rule: AccessRule,
 }
 
 pub const EPOCH_MANAGER_UPDATE_VALIDATOR_IDENT: &str = "update_validator";
 
-#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
+#[derive(
+    Debug, Clone, Eq, PartialEq, ScryptoSbor, ManifestCategorize, ManifestEncode, ManifestDecode,
+)]
 pub enum UpdateValidator {
     Register(EcdsaSecp256k1PublicKey, Decimal),
     Unregister,
 }
 
-#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
+#[derive(
+    Debug, Clone, Eq, PartialEq, ScryptoSbor, ManifestCategorize, ManifestEncode, ManifestDecode,
+)]
 pub struct EpochManagerUpdateValidatorInput {
     pub validator_address: ComponentAddress,
     pub update: UpdateValidator,
 }
 
 pub const VALIDATOR_REGISTER_IDENT: &str = "register";
 
-#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
+#[derive(Debug, Clone, Eq, PartialEq, Sbor)]
```

Comment:
Should we just use all `ManifestCategorize, ManifestEncode, ManifestDecode` for all inputs. This will make it easier to implement the `native_blueprint!` macro to be used.

Reply:
We can't add `ManifestCategorize`, `ManifestEncode`, `ManifestDecode`  to every input, especially the one that contains `Bucket`, `Proof` or `KeyValueStore`.

Although, I want to understand more how manifest encoding is used in `native_blueprint!` macro though. Are you saying you intended to make every input struct work for both Scrypto invocation and manifest call args?